app-backend/app/src/main/scala/bucket/Routes.scala

```diff
@@ -95,6 +96,21 @@ trait BucketRoutes extends Authentication
                   }
                 }
               }
+            } ~
+            authenticate { user =>
+              post {
+                entity(as[BulkSceneOperation]) { bulkOp =>
+                  bulkOp.action match {
+                    case "upsert" => onSuccess(Buckets.addScenesToBucket(bulkOp.sceneIds, bucketId)) {
+                      _ => complete(StatusCodes.Created)
+                    }
+                    case "delete" => onSuccess(Buckets.deleteScenesFromBucket(bulkOp.sceneIds, bucketId)) {
+                      _ => complete(StatusCodes.NoContent)
+                    }
+                    case _ => throw new IllegalArgumentException(s"Unrecognized bulk operation ${bulkOp.action}")
```

Comment:
`IllegalArgumentException` seems to be the emerging standard for saying `400 Bad Request`. I should probably add a `UserErrorHandler` trait to this route. I wonder if `${bulkOp.action}` is extraneous and can be captured in the `case` statement.